lib/platform.h

```diff
@@ -43,24 +43,24 @@ namespace cppcheck {
         static long long min_value(int bit) {
             if (bit >= 64)
                 return LLONG_MIN;
-            return -(1LL << (bit-1));
+            return -(1LL << (static_cast<long long>(bit - 1)));
```

Comment:
heh.. I still do not understand the point of this.

This little program gives me no output:
```
int main() {
    for (int bit = 1; bit <= 62; ++bit) {
        long long l1 = -(1LL << (bit-1));
        long long l2 = -(1LL << (bit-1));
        if (l1 != l2)
            std::cout << "mismatch:" << bit << std::endl;
    }
    return 0;
}
```

Reply:
Passing in an int and coercing it out to long long after some manipulation. Clang made noise about. Disconcerting that you test program is failing.

Reply:
> Disconcerting that you test program is failing.

Are you saying that you compile this on your computer and run it... and you get some "mismatch" output? what bit value does it complain about?

Reply:
When I compile this little test program with clang, my version of clang does not say much:
```
danielm@edge:~/cppcheck$ clang++ -Weverything -o 1 1.cpp
1.cpp:5:9: warning: 'long long' is a C++11 extension [-Wc++11-long-long]
        long long l1 = -(1LL << (bit-1));
        ^
1.cpp:5:26: warning: 'long long' is a C++11 extension [-Wc++11-long-long]
        long long l1 = -(1LL << (bit-1));
                         ^
1.cpp:6:9: warning: 'long long' is a C++11 extension [-Wc++11-long-long]
        long long l2 = -(1LL << static_cast<long long>(bit-1));
        ^
1.cpp:6:26: warning: 'long long' is a C++11 extension [-Wc++11-long-long]
        long long l2 = -(1LL << static_cast<long long>(bit-1));
                         ^
1.cpp:6:45: warning: 'long long' is a C++11 extension [-Wc++11-long-long]
        long long l2 = -(1LL << static_cast<long long>(bit-1));
                                            ^

```

Do you get something more interesting?

Reply:
Just tried it here w/ clang. Even stretched it out to range 1..64. Matches all the way.

Reply:
`clang++ --std=c++11 main.cpp -o testlong -Weverything -Wno-c++98-compat-pedantic`
silent....
with std::hex from iomanip, l1 and l2 count down from `ffffffffffffffff` to `8000000000000000`.
Or in decimal -1 to -9223372036854775808.

Reply:
This is just a bit shift with right fill of -1 or 0xF. Nothing fancy.

Reply:
```
#include <iostream>
#include <iomanip>

int main() {
    for (int bit = 1; bit <= 64; ++bit)
    {
        long long l1 = -(1LL << (bit-1));
        long long l2 = -(1LL << static_cast<long long>(bit-1));
        std::cout << "bit count: " << std::setw(4) << bit
                  << "\tl1: " << std::setw(20) << std::hex << l1
                  << "\tl2: " << std::setw(20) << std::hex << l2
                  << std::endl;
        if (l1 != l2)
            std::cout << "mismatch:" << bit << std::endl;
    }
    return 0;
}
```
gcc is also quiet and program works. Same results as clang.